### DISCUSSION TITLE
[Feature] Multiple dates search (API and GUI)

### DISCUSSION THREAD

**Discussion by nilsburg** [immich-app/immich#12046] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi,
I would like to be able to search photos with multiple dates (like "pictures from this this week through several years").
For example to be able to search pictures between multiple __takenAfter__ and __takenBefore__
Example using API:
```
curl -L '/api/search/metadata' \
-H 'Content-Type: application/json' \
-H 'Accept: application/json' \
-d '{
  dates: [
  	{"takenAfter":"2024-08-19", takenBefore:"2024-08-25"},
  	{"takenAfter":"2023-08-19", takenBefore:"2023-08-25"},
  	{"takenAfter":"2022-08-19", takenBefore:"2022-08-25"},
  ]
}'
```
I haven't tried to put multiple dates with the API, but since it is not documented I guess it is not implemented.

I've been looking if this was already possible but couldn't find it.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by C-Otto** [comment]
Why don't you just run multiple individual queries and combine the responses yourself?

**Reply by nilsburg** [reply]
Yes, that is one solution using the API, but using the UI it won't be possible this way. 
But for my use case I can run multiple queries.
Thanks.
